tsdb/engine/tsm1/reader.go

```diff
@@ -995,43 +990,35 @@ func (a *indexEntries) Append(entry ...*IndexEntry) {
 	a.entries = append(a.entries, entry...)
 }
 
-func (a *indexEntries) MarshalBinary() (b []byte, err error) {
-	for _, entry := range a.entries {
-		b = append(b, u64tob(uint64(entry.MinTime.UnixNano()))...)
-		b = append(b, u64tob(uint64(entry.MaxTime.UnixNano()))...)
-		b = append(b, u64tob(uint64(entry.Offset))...)
-		b = append(b, u32tob(entry.Size)...)
+func (a *indexEntries) MarshalBinary() ([]byte, error) {
+	buf := make([]byte, len(a.entries)*indexEntrySize)
+
+	for i, entry := range a.entries {
+		entry.AppendTo(buf[indexEntrySize*i:])
 	}
-	return b, nil
-}
 
-func (a *indexEntries) Write(w io.Writer) error {
-	for _, entry := range a.entries {
-		_, err := w.Write(u64tob(uint64(entry.MinTime.UnixNano())))
-		if err != nil {
-			return err
-		}
-		_, err = w.Write(u64tob(uint64(entry.MaxTime.UnixNano())))
-		if err != nil {
-			return err
-		}
+	return buf, nil
+}
 
-		_, err = w.Write(u64tob(uint64(entry.Offset)))
-		if err != nil {
-			return err
-		}
+func (a *indexEntries) WriteTo(w io.Writer) (total int64, err error) {
```

Comment:
I think that would work.  `AppendTo` doesn't feel like the right name, but I can't think of a better one.